4-Beyond-Classical-Search/hillClimbing.js

```diff
@@ -1,22 +1,95 @@
-var HillClimber = function(){
-	// Should the robot go left, right or stay where it is
-	this.DECISIONS = {
-		LEFT:-1,
-		RIGHT:1,
-		STAY:0
-	};
-	// Decide where to go based on height
-	this.decide = function(left,current,right){
-		if(left == right && left > current){
-			this.currentDecision = Math.random() > .5 ?
-			this.DECISIONS.LEFT: this.DECISIONS.RIGHT;
-		} else if(right == null || left > right){
-			this.currentDecision = this.DECISIONS.LEFT;
-		} else if (left == null || right > left) {
-			this.currentDecision = this.DECISIONS.RIGHT;
-		} else {
-			this.currentDecision = this.DECISIONS.STAY;
+class Hill {
+
+	getRandomValues() {
+		this.spacing = 10;
+		this.random = [];
+		for(let i = 0; i < this.spacing+1;i++) {
+			this.random.push(Math.floor(Math.random()*(100-5 + 1) + 5));
 		}
-		return this.currentDecision;
-	};
-};
+	}
+	f(i) {
+		let x1 = Math.floor(i/this.spacing);
+		let x2 = Math.ceil(i/this.spacing);
+		let interAmount = (i%this.spacing)/this.spacing;
+		let n1 = this.random[x1];
+		let n2 = this.random[x2];
+		let k = 0.5 - 0.5*Math.cos(interAmount*Math.PI);
+	  let ans = (1-k)*n1 + k*n2;
+		return ans;
+	}
+
+  generateRandomHill() {
+		this.getRandomValues();
+    let states = [];
```

Comment:
There's some weird indentation here. It looks like it is a tab size issue in your editor. If you can, set the indentation to 2 spaces, but the physical tab size to 8 spaces to match github and other applications. Or set it to always use spaces.